File: 20220727/모범답안/20220727_02.js/CountEvenEachNumber_test.js
import assert from 'assert'
import solution from './CountEvenEachNumber.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const n = 2875812

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(4, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const n = 1

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(0, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const n = 2

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(1, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const n = 876382

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(4, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const n = 158

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(1, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const n = 96273849

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(4, result)
    done();
  });
})

File: 20220727/모범답안/20220727_02.js/CountEvenEachNumber_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.supereasy.CountEvenEachNumber.CountEvenEachNumber import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        n = 2875812

        # when
        result = solution(n)

        # then
        self.assertEqual(4, result)

    def test_2(self):
        # given
        n = 1

        # when
        result = solution(n)

        # then
        self.assertEqual(0, result)

    def test_3(self):
        # given
        n = 2

        # when
        result = solution(n)

        # then
        self.assertEqual(1, result)

    def test_4(self):
        # given
        n = 876382

        # when
        result = solution(n)

        # then
        self.assertEqual(4, result)

    def test_5(self):
        # given
        n = 158

        # when
        result = solution(n)

        # then
        self.assertEqual(1, result)

    def test_6(self):
        # given
        n = 96273849

        # when
        result = solution(n)

        # then
        self.assertEqual(4, result)


if __name__ == '__main__':
    unittest.main()
